src/alpaka_job_coverage/filter_software_dependency.py

```diff
@@ -136,13 +175,18 @@ def software_dependency_filter(row: List) -> bool:
         and row_check_name(row, DEVICE_COMPILER, "==", HIPCC)
         and row_check_backend_version(row, ALPAKA_ACC_GPU_HIP_ENABLE, "!=", OFF_VER)
     ):
+        reason(output, "all rocm images are Ubuntu 20.04 based")
         return False
 
     # a bug in CMAKE 3.18 avoids the correct usage of the variable CMAKE_CUDA_ARCHITECTURE if the
     # CUDA compiler is Clang++
     if row_check_name(row, DEVICE_COMPILER, "==", CLANG_CUDA) and row_check_version(
         row, CMAKE, "<", "3.19"
     ):
+        reason(
+            output,
+            "CMake 3.18 and older does not support Clang as CUDA compiler correctly.",
```

Comment:
```suggestion
            "CMake versions <= 3.18 do not support clang-cuda correctly.",
```

This is also the first time I noticed a `.` at the end of a sentence. This should be uniform across all messages.